skimage/restoration/tests/test_restoration.py

```diff
@@ -11,46 +12,60 @@
 test_img = util.img_as_float(camera())
 
 
-def test_wiener():
-    psf = np.ones((5, 5)) / 25
+@testing.parametrize('dtype', [np.float32, np.float64])
+def test_wiener(dtype):
+    psf = np.ones((5, 5), dtype=dtype) / 25
     data = convolve2d(test_img, psf, 'same')
     np.random.seed(0)
     data += 0.1 * data.std() * np.random.standard_normal(data.shape)
+    data = data.astype(dtype, copy=False)
     deconvolved = restoration.wiener(data, psf, 0.05)
+    assert deconvolved.dtype == dtype
 
     path = fetch('restoration/tests/camera_wiener.npy')
-    np.testing.assert_allclose(deconvolved, np.load(path), rtol=1e-3)
+    atol = 1e-5 if dtype == np.float32 else 0
```

Comment:
how come, we can afford 0 tolerance for all non-FP32 types? FP16 will definitely need a larger one. For FP64 it might work, but a hard 0 seems like a strong claim.